### DISCUSSION TITLE
Repeated offenders

### DISCUSSION THREAD

**Discussion by michelkluger** [laurentS/slowapi#196] [Q&A] [answered]
If I am trying to limit a user for abusing the limits repeatedly,

What would be a good way to make limits increasingly smaller for that user, or times without access increasingly bigger?

any suggestions?

**Comment by laurentS** [comment, marked as answer]
I think you could use the usual limiter with a dynamic limit value, as shown in [this test](https://github.com/laurentS/slowapi/blob/master/tests/test_fastapi_extension.py#L241).

I'd suggest you look at https://github.com/laurentS/slowapi/issues/13#issuecomment-1860629613 and adapt that logic to provide a limit on each call. You will probably need some sort of data storage in your backend to keep track of each user's state with respect to the limit.
